scripts/python/services/GithubService.py

```diff
@@ -0,0 +1,15 @@
+import logging
+
+from github import Github
+
+
+class GithubService:
+    def __init__(self, org_token: str, organisation_name: str) -> None:
```

Comment:
Resisted the temptation here to enforce a [Singleton Pattern](https://en.wikipedia.org/wiki/Singleton_pattern).

Though I did have a read about how to do this in python and there seemed like a few options described by this [stackoverflow answer](https://stackoverflow.com/questions/6760685/creating-a-singleton-in-python)

Decided I didn't know enough about python to choose appropriately, instead I've opted to leave it out for the time being - and maybe implement this at a later date 😄

Reply:
I remember singleton pattern was one to avoid, plenty of stuff online why, such as [Open / Closed Principle Violation](https://www.davidtanzer.net/david's%20blog/2016/03/14/6-reasons-why-you-should-avoid-singletons.html)

Reply:
Prob wont need a singleton pattern class, if create the object at the start of the application and associate with the other classes at runtime

Reply:
Skimmed down the article, but do agree with the points in there. I agree that the pattern comes with a fair few downsides, the main one for me is the difficulty it adds to testing 🧪 

I'd say the Singleton is still a good solution for when there's a high risk of duplicating an object and you need strict gatekeeping logic. Though the singleton is probably a bandage solution to a deeper problem in that sense 🩹 😂